File: year2/mpp/java/MotorcycleContest/src/main/java/domain/validators/RaceValidator.java
package domain.validators;

import domain.ValidationException;
import domain.entities.Race;
import org.springframework.stereotype.Component;

@Component
public class RaceValidator implements Validator<Race> {
    @Override
    public void validate(Race race) throws ValidationException {
        if (race.getCapacity() == null) {
            throw new ValidationException("All race fields must be not null");
        }
        if (race.getCapacity() < 50 || race.getCapacity() > 2000) {
            throw new ValidationException("Race capacity is invalid");
        }
    }
}

File: year2/mpp/c#/MotorcycleGrpcServer/MotorcycleCommon/Validators/RaceValidator.cs
using MotorcycleContest.Domain.Entities;

namespace MotorcycleContest.Domain.Validators
{
    public class RaceValidator : IValidator<Race>
    {
        public void Validate(Race race)
        {
            if (race.Capacity.Equals(-1))
            {
                throw new ValidationException("null fields");
            }
            if (race.Capacity < 50 || race.Capacity > 2000)
            {
                throw new ValidationException("capacity is invalid");
            }
        }
    }
}
